packages/admin/src/targetGroups/TargetGroupForm.tsx

```diff
@@ -27,41 +28,54 @@ import { createTargetGroupMutation, targetGroupFormQuery, updateTargetGroupMutat
 import {
     GQLCreateTargetGroupMutation,
     GQLCreateTargetGroupMutationVariables,
-    GQLTargetGroupFormFragment,
     GQLTargetGroupFormQuery,
     GQLTargetGroupFormQueryVariables,
     GQLUpdateTargetGroupMutation,
     GQLUpdateTargetGroupMutationVariables,
 } from "./TargetGroupForm.gql.generated";
 
-type FormValues = GQLTargetGroupFormFragment;
+export interface EditTargetGroupFinalFormValues {
+    title: string;
+    [key: string]: unknown;
+}
 
 interface FormProps {
     id?: string;
     scope: ContentScopeInterface;
+    additionalFormFields?: React.ReactNode;
+    nodeFragment?: { name: string; fragment: DocumentNode };
+    dataToInitialValues?: (values?: EditTargetGroupFinalFormValues) => EditTargetGroupFinalFormValues;
 }
 
-export function TargetGroupForm({ id, scope }: FormProps): React.ReactElement {
+export function TargetGroupForm({ id, scope, additionalFormFields, dataToInitialValues, nodeFragment }: FormProps): React.ReactElement {
     const stackApi = useStackApi();
     const client = useApolloClient();
     const mode = id ? "edit" : "add";
-    const formApiRef = useFormApiRef<FormValues>();
+    const formApiRef = useFormApiRef<EditTargetGroupFinalFormValues>();
     const stackSwitchApi = useStackSwitchApi();
 
+    const targetGroupFormFragment = gql`
+        fragment TargetGroupForm on TargetGroup {
+            title
+            ${nodeFragment ? "...".concat(nodeFragment?.name) : ""}
+        }
+        ${nodeFragment?.fragment ?? ""}
+    `;
+
     const { data, error, loading, refetch } = useQuery<GQLTargetGroupFormQuery, GQLTargetGroupFormQueryVariables>(
-        targetGroupFormQuery,
+        targetGroupFormQuery(targetGroupFormFragment),
         id ? { variables: { id } } : { skip: true },
     );
 
-    const initialValues = React.useMemo<Partial<FormValues>>(
-        () =>
-            data?.targetGroup
-                ? {
-                      title: data.targetGroup.title,
-                  }
-                : {},
-        [data],
-    );
+    const initialValues = React.useMemo<Partial<EditTargetGroupFinalFormValues>>(() => {
+        let additionalInitialValues = {};
+
+        if (dataToInitialValues) {
+            additionalInitialValues = dataToInitialValues(data?.targetGroup);
+        }
+
+        return data?.targetGroup ? { title: data.targetGroup.title, ...additionalInitialValues } : { ...additionalInitialValues };
```

Comment:
```suggestion
        return data?.targetGroup ? { title: data.targetGroup.title, ...additionalInitialValues } : additionalInitialValues ;
```

Reply:
adapted in [8cdbcf4](https://github.com/vivid-planet/comet-brevo-module/pull/30/commits/8cdbcf48df375f6a0b5a8125fe8fe9f594da1201)